hawkular-metrics/hawkular-metrics-wrapper.sh

```diff
@@ -91,16 +91,46 @@ fi
 sed -i "s|<!-- ##ADDITIONAL LOGGERS## -->|$additional_loggers|g" ${JBOSS_HOME}/standalone/configuration/standalone.xml
 
 # Setup the truststore so that it will accept the OpenShift cert
+KEYTOOL_COMMAND=/usr/lib/jvm/java-1.8.0/jre/bin/keytool
 HAWKULAR_METRICS_AUTH_DIR=$HAWKULAR_METRICS_DIRECTORY/auth
 mkdir $HAWKULAR_METRICS_AUTH_DIR
 pushd $HAWKULAR_METRICS_AUTH_DIR
 
-cp $KEYSTORE hawkular-metrics.keystore
-cp $TRUSTSTORE hawkular-metrics.truststore
+if [ ! -n "$KEYSTORE" ]; then
+    ## no keystore specified, so, we set one up based on a cert that we expect to exist at `/hawkular-metrics-certs`
+    KEYSTORE_DIR=${KEYSTORE_DIR:-"${HAWKULAR_METRICS_AUTH_DIR}"}
+    KEYSTORE=${KEYSTORE:-"${KEYSTORE_DIR}/hawkular-metrics.keystore"}
+    KEYSTORE_PASSWORD=${KEYSTORE_PASSWORD:-$(head /dev/urandom -c 512 | tr -dc A-Z-a-z-0-9 | head -c 17)}
+    SERVICE_CERT=${SERVICE_CERT:-"/hawkular-metrics-certs/tls.crt"}
+    SERVICE_CERT_KEY=${SERVICE_CERT_KEY:-"/hawkular-metrics-certs/tls.key"}
+    PKCS12_FILE=${PKCS12_FILE:-"${KEYSTORE_DIR}/hawkular-metrics.pkcs12"}
+    SERVICE_ALIAS=${SERVICE_ALIAS:-"hawkular-metrics"}
+
+    echo "Creating the Hawkular Metrics keystore from the Secret's cert data"
+    openssl pkcs12 -export -in ${SERVICE_CERT} -inkey ${SERVICE_CERT_KEY} -out ${PKCS12_FILE} -name ${SERVICE_ALIAS} -noiter -nomaciter -password pass:${KEYSTORE_PASSWORD}
+    if [ $? != 0 ]; then
+        echo "Failed to create a PKCS12 certificate file with the service-specific certificate. Aborting."
+        exit 1
+    fi
+    echo "Converting the PKCS12 keystore into a Java Keystore"
+    ${KEYTOOL_COMMAND} -v -importkeystore -srckeystore ${PKCS12_FILE} -srcstoretype PKCS12 -destkeystore ${KEYSTORE} -deststoretype JKS -deststorepass ${KEYSTORE_PASSWORD} -srcstorepass ${KEYSTORE_PASSWORD}
+    if [ $? != 0 ]; then
+        echo "Failed to create a Java Keystore file with the service-specific certificate. Aborting."
+        exit 1
+    fi
+else
+    cp $KEYSTORE hawkular-metrics.keystore
+fi
+
+if [ -n "$TRUSTSTORE" ]; then
```

Comment:
We should be generating the truststore and not give a user an option to pass one in